Exit when unable get self-sign cert

jhadvig opened the pull request:
@benjaminapetersen PTAL

benjaminapetersen commented:
@jhadvig So should we force a fail, or do we want to support the possibility of allowing the OAuth flow?  Maybe with an environment variable `--OAUTH` we could skip this check?

jhadvig commented:
@benjaminapetersen definitelly. Opened this PR to discuss the all the possible flow that we want cover, from obtaining self-sign certs, to oauth...
Lets chat to get a clear picture to solve this :)

jhadvig commented:
@benjaminapetersen let's talk today regarding this, so we align on the different flows when we will need the self-sign certificate and we won't need it.

jhadvig commented:
@benjaminapetersen I've updated the PR. Refactored it a bit so now obtaining the cert will be optional (by default will be obtained), since when in use in production cluster we probably wont need to get the cert and store it.
PTAL

benjaminapetersen commented on smoke-test.sh:
```diff
@@ -15,11 +15,18 @@ host_with_port=$(echo "$url" | sed 's/\/.*$//')
-os::log::info "Storing obtained certificate"
-certutil -d "sql:$HOME/.pki/nssdb" -A -n console -t Pu,, -i /tmp/console-e2e.pem
+if [ "$SELF_SIGN_CERT" = true ]; then
+  os::log::info "Getting certificate for ${CONSOLE_URL}"
```
Shall we update the log to "Getting self signed certificate...." just to help ensure this is super clear?

benjaminapetersen commented:
@jhadvig nice
